'Hash Sum Mismatch' bug on mac when building docker

**Operating system name and version:** MacOS 26.0.1

**Any details about your local setup that might be helpful in troubleshooting:**
- Device: MacBook Air M1 2020 (Apple Silicon, arm64)
- Environment: Terminal in VS Code. Base conda environment activated. Docker installed. Under the "main" branch in "FNLCR-DMAP/SPAC_Shiny" repo.
- Network: Without proxies (no VPNs). This bug appears under both Purdue campus' network and my home's network.
- Image: This bug appears both for the previous Python 3.9.13 image and the current Python 3.9.19-slim-bookworm image.

**Detailed steps to reproduce the bug:**
1. Open VS Code's Terminal and checkout the updated "main" branch in "FNLCR-DMAP/SPAC_Shiny" repo locally.
2. Run `make run` in the terminal. **It then crashed in "Step 3: RUN apt-get update && apt-get install"**  when upgrading packages with (possibly multiple) 'Hash Sum Mismatch' error message(s).
3. I tried several times and the 'Hash Sum Mismatch' error appeared for different packages. Whichever packages affected, this mismatch bug always exists.

**Example of an error message:**

```
(base) bojohnzhang@pal-nat186-51-219 SPAC_Shiny-Purdue-2025 % make run
Building SPAC Shiny Docker image...
docker build -t spac-shiny .
[+] Building 42.2s (8/11)                                                                docker:desktop-linux
 => [internal] load build definition from Dockerfile                                                     0.0s
 => => transferring dockerfile: 679B                                                                     0.0s
 => [internal] load metadata for docker.io/library/python:3.9.19-slim-bookworm  
 => CACHED [2/6] WORKDIR /app                                                                            0.0s
 => ERROR [3/6] RUN apt-get update && apt-get install -y     gcc     g++     git     libhdf5-dev     p  41.5s
------
 > [3/6] RUN apt-get update && apt-get install -y     gcc     g++     git     libhdf5-dev     pkg-config     && rm -rf /var/lib/apt/lists/*:
0.246 Get:1 http://deb.debian.org/debian bookworm InRelease [151 kB]
...
5.458 The following packages will be upgraded:
3.459   gcc-12-base libc-bin libc6 libgcc-s1 libssl3 libstdc++6 openssl perl-base
3.504 8 upgraded, 119 newly installed, 0 to remove and 34 not upgraded.
3.504 Need to get 106 MB of archives.
3.504 After this operation, 432 MB of additional disk space will be used.
3.504 Get:1 http://deb.debian.org/debian bookworm/main arm64 perl-base arm64 5.36.0-7+deb12u3 [1475 kB]
...
4.964 Get:26 http://deb.debian.org/debian bookworm/main arm64 binutils arm64 2.40-2 [64.9 kB]
4.966 Get:27 http://deb.debian.org/debian bookworm/main arm64 libisl23 arm64 0.25-1.1 [610 kB]
5.015 Err:27 http://deb.debian.org/debian bookworm/main arm64 libisl23 arm64 0.25-1.1
5.015   Hash Sum mismatch
5.015   Hashes of expected file:
5.015    - SHA256:e7232e299714254a9473d7eacdf6879eedf02b950c87da145f5d605800710c6c
5.015    - MD5Sum:cf54a84feab8ba9487989cc9f53f22d6 [weak]
5.015    - Filesize:609984 [weak]
5.015   Hashes of received file:
5.015    - SHA256:a228723d3f4af098759274abcf3fff3d5138ed426cc249bc2f60f005c4831fb9
5.015    - MD5Sum:80942b90c7bd14f67abac2dd8fb80a74 [weak]
5.015    - Filesize:609984 [weak]
5.015   Last modification reported: Sat, 20 Jan 2024 17:23:55 +0000
5.031 Ign:28 http://deb.debian.org/debian bookworm/main arm64 libmpfr6 arm64 4.2.0-1
5.064 Get:29 http://deb.debian.org/debian bookworm/main arm64 libmpc3 arm64 1.3.1-1 [49.2 kB]
5.078 Get:30 http://deb.debian.org/debian bookworm/main arm64 cpp-12 arm64 12.2.0-14+deb12u1 [8228 kB]
5.446 Get:31 http://deb.debian.org/debian bookworm/main arm64 cpp arm64 4:12.2.0-3 [6832 B]
5.446 Get:32 http://deb.debian.org/debian bookworm/main arm64 fonts-dejavu-core all 2.37-6 [1068 kB]
5.497 Err:32 http://deb.debian.org/debian bookworm/main arm64 fonts-dejavu-core all 2.37-6
5.497   Hash Sum mismatch
5.497   Hashes of expected file:
5.497    - SHA256:8892669e51aab4dc56682c8e39d8ddb7d70fad83c369344e1e240bf3ca22bb76
5.497    - MD5Sum:755d6c59d57accb3000de0cdba40918d [weak]
5.497    - Filesize:1067728 [weak]
5.497   Hashes of received file:
5.497    - SHA256:7221c47025de1a31028e9a82afef0c141fc1df295f2cec7532773d94bf3efaf1
5.497    - MD5Sum:933ca45d169e9428cad9a8d9ea3bbba2 [weak]
5.497    - Filesize:1067728 [weak]
5.497   Last modification reported: Fri, 10 Mar 2023 09:24:57 +0000
35.56 Get:33 http://deb.debian.org/debian bookworm/main arm64 fontconfig-config arm64 2.14.1-4 [315 kB]
...
38.26 Get:61 http://deb.debian.org/debian bookworm/main arm64 libpsl5 arm64 0.21.2-1 [58.6 kB]
38.26 Err:61 http://deb.debian.org/debian bookworm/main arm64 libpsl5 arm64 0.21.2-1
38.26   File has unexpected size (68620 != 58600). Mirror sync in progress? [IP: 151.101.2.132 80]
38.26   Hashes of expected file:
38.26    - SHA256:90a68e4a6d9eb95abee32595762dbf67974f13c158e4ce4a05c7c2b54440db73
38.26    - MD5Sum:04b520a58484299c870805248de377d7 [weak]
38.26    - Filesize:58600 [weak]
38.29 Get:62 http://deb.debian.org/debian bookworm/main arm64 librtmp1 arm64 2.4+20151223.gitfa8646d.1-2+b2 [59.4 kB]
...
41.42 Get:127 http://deb.debian.org/debian bookworm/main arm64 xauth arm64 1:1.1.2-1 [35.5 kB]
41.44 Get:28 http://deb.debian.org/debian bookworm/main arm64 libmpfr6 arm64 4.2.0-1 [600 kB]
41.48 Fetched 106 MB in 38s (2784 kB/s)
41.48 E: Failed to fetch http://deb.debian.org/debian/pool/main/i/isl/libisl23_0.25-1.1_arm64.deb  Hash Sum mismatch
41.48    Hashes of expected file:
41.48     - SHA256:e7232e299714254a9473d7eacdf6879eedf02b950c87da145f5d605800710c6c
41.48     - MD5Sum:cf54a84feab8ba9487989cc9f53f22d6 [weak]
41.48     - Filesize:609984 [weak]
41.48    Hashes of received file:
41.48     - SHA256:a228723d3f4af098759274abcf3fff3d5138ed426cc249bc2f60f005c4831fb9
41.48     - MD5Sum:80942b90c7bd14f67abac2dd8fb80a74 [weak]
41.48     - Filesize:609984 [weak]
41.48    Last modification reported: Sat, 20 Jan 2024 17:23:55 +0000
41.48 E: Failed to fetch http://deb.debian.org/debian/pool/main/f/fonts-dejavu/fonts-dejavu-core_2.37-6_all.deb  Hash Sum mismatch
41.48    Hashes of expected file:
41.48     - SHA256:8892669e51aab4dc56682c8e39d8ddb7d70fad83c369344e1e240bf3ca22bb76
41.48     - MD5Sum:755d6c59d57accb3000de0cdba40918d [weak]
41.48     - Filesize:1067728 [weak]
41.48    Hashes of received file:
41.48     - SHA256:7221c47025de1a31028e9a82afef0c141fc1df295f2cec7532773d94bf3efaf1
41.48     - MD5Sum:933ca45d169e9428cad9a8d9ea3bbba2 [weak]
41.48     - Filesize:1067728 [weak]
41.48    Last modification reported: Fri, 10 Mar 2023 09:24:57 +0000
41.48 E: Failed to fetch http://deb.debian.org/debian/pool/main/libp/libpsl/libpsl5_0.21.2-1_arm64.deb  File has unexpected size (68620 != 58600). Mirror sync in progress? [IP: 151.101.2.132 80]
41.48    Hashes of expected file:
41.48     - SHA256:90a68e4a6d9eb95abee32595762dbf67974f13c158e4ce4a05c7c2b54440db73
41.48     - MD5Sum:04b520a58484299c870805248de377d7 [weak]
41.48     - Filesize:58600 [weak]
41.48 E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
------
Dockerfile:8
--------------------
   7 |     # Install system dependencies needed for scientific packages
   8 | >>> RUN apt-get update && apt-get install -y \
   9 | >>>     gcc \
  10 | >>>     g++ \
  11 | >>>     git \
  12 | >>>     libhdf5-dev \
  13 | >>>     pkg-config \
  14 | >>>     && rm -rf /var/lib/apt/lists/*
  15 |     
--------------------
ERROR: failed to build: failed to solve: process "/bin/sh -c apt-get update && apt-get install -y     gcc     g++     git     libhdf5-dev     pkg-config     && rm -rf /var/lib/apt/lists/*" did not complete successfully: exit code: 100
make: *** [build] Error 1
(base) bojohnzhang@pal-nat186-51-219 SPAC_Shiny-Purdue-2025 %
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -4,6 +4,11 @@ FROM python:3.9.19-slim-bookworm
 # Set working directory
 WORKDIR /app
 
+# Fix 'Hash Sum Mismatch' bug for mac device
+RUN echo "Acquire::http::Pipeline-Depth 0;" > /etc/apt/apt.conf.d/99custom && \
+    echo "Acquire::http::No-Cache true;" >> /etc/apt/apt.conf.d/99custom && \
+    echo "Acquire::BrokenProxy    true;" >> /etc/apt/apt.conf.d/99custom
+
 # Install system dependencies needed for scientific packages
 RUN apt-get update && apt-get install -y \
     gcc \

